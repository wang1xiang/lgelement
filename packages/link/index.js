import WangLink from './src/link.vue'

WangLink.install = Vue => {
  Vue.component(WangLink.name, WangLink)
}

export default WangLink
