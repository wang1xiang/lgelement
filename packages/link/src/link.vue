<template>
  <a
    :href="disabled ? null : href"
    :class="[disabled && 'disabled', !underlined && 'underlined']"
  >
    <slot></slot
  ></a>
</template>

<script>
export default {
  name: 'WangLink',
  props: {
    href: {
      type: String,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    underlined: {
      type: Boolean,
      default: false,
    },
  },
}
</script>

<style scoped>
.disabled {
  cursor: not-allowed;
}
.underlined {
  text-decoration: none;
}
</style>
