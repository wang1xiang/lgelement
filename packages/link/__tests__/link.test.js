import { mount } from '@vue/test-utils'
import link from '../src/link.vue'

describe('Wang-Link', () => {
  test('link-disabled-underlined', () => {
    // 挂载组件 只是内存中的挂载 返回一个包裹器
    const wrapper = mount(link, {
      propsData: {
        disabled: true,
      },
    })
    expect(wrapper.html()).toContain('class="disabled underlined"')
  })
  test('link-disabled', () => {
    const wrapper = mount(link, {
      propsData: {
        disabled: true,
        underlined: true,
      },
    })
    expect(wrapper.html()).toContain('class="disabled"')
  })
  test('link-a', () => {
    const wrapper = mount(link, {
      propsData: {
        href: 'www.baidu.com',
      },
    })
    expect(wrapper.props('href')).toBe('www.baidu.com')
  })
  test('link-snapshot', () => {
    const wrapper = mount(link, {
      propsData: {
        href: 'www.baidu.com',
        disabled: true,
      },
    })
    // 快照 第一次运行会将wrapper.vm.$el的内容存储在./__snapshots__/link.test.js.snap中
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
