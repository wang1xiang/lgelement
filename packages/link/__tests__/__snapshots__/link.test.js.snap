// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wang-Link link-snapshot 1`] = `
<a
  class="disabled underlined"
/>
`;
